### DISCUSSION TITLE
Turbolinks No Longer Developed

### DISCUSSION THREAD

**Discussion by billksun** [digitallyinduced/ihp#1795] [General]
The Turbolink GitHub repo has been [archived](https://github.com/turbolinks/turbolinks). Are there any plans to replace this dependency, or take over the maintenance? I am mainly concern with keeping up-to-date with all the NPM security bumps.

**Comment by MonaMayrhofer** [comment]
It should be mentioned that htmx, which [integrates very well with ihp](https://ihp.digitallyinduced.com/Guide/htmx-and-hyperscript.html) anyways, has a feature that pretty much replaces turbolinks, called [Boosting](https://htmx.org/docs/#boosting).

Just add htmx to your scripts and then add `hx-boost="true"` to the body in your Layout, and all anchors are promoted to ajax requests that swap out the page. 

Also, dont forget to remove the turbolinks scripts 

Htmx also [supports morphdom](https://htmx.org/docs/#morphing), so even `turbolinksMorphdom.js` is covered.

A replacement for `turbolinksInstantClick.js` would be the [htmx-preload extensions](https://htmx.org/extensions/preload/)

**Reply by mpscholten** [reply]
Maybe we should remove turbolinks from the default IHP boilerplate then and then recommend using htmx

**Reply by MonaMayrhofer** [reply]
Its definitely worth looking into!
Htmx comes with a lot of other stuff too tho, so shipping it by default would probably make the template more opinionated

But stripping turbolinks and just recommending htmx sounds like a good idea😁

**Reply by amitaibu** [reply]
I think one of the big appeals of IHP is that it feels like SPA out of the box. So I think that if we take Turbolinks out, we should put htmx in and not just recommend it. Or said differently, we can have the docs be about how to _remove_ htmx.
